src/core/sign.ml

```diff
@@ -149,11 +149,12 @@ let link : t -> unit = fun sign ->
   in
   PathMap.iter gn !(sign.sign_deps);
   sign.sign_builtins := StrMap.map link_symb !(sign.sign_builtins);
-  let hn (s,h) = (link_symb s, h) in
-  sign.sign_unops := StrMap.map hn !(sign.sign_unops);
-  sign.sign_binops := StrMap.map hn !(sign.sign_binops);
+  let lsy (sym, h) = link_symb sym, h in
+  sign.sign_syntax :=
+    (* Keys of the mapping are linked *)
+    SymMap.to_seq !(sign.sign_syntax) |>
+    Seq.map lsy |> SymMap.of_seq;
```

Comment:
Why not using a SymMap.fold instead?

Reply:
Well it's really similar. I don't have any real preference.

Reply:
And I think neither solution is better than the other, performance wise.